CMake should fail if no collision library available

Installing v6.7.0 on a new machine (Arch Linux) and forgot to install FCL.

CMake warned about missing FCL and ODE collision, but didn't exit with an error.

Running `make`, of course, failed.

I think it makes sense to have `cmake` exit with error if it know compilation is doomed to fail.

`make` output:
```
In file included from /home/egordon/Workspace/dart/dart/collision/fcl/BackwardCompatibility.cpp:33:
/home/egordon/Workspace/dart/dart/collision/fcl/BackwardCompatibility.hpp:49:10: fatal error: fcl/config.h: No such file or directory
 #include <fcl/config.h>
          ^~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [dart/CMakeFiles/dart.dir/build.make:1493: dart/CMakeFiles/dart.dir/collision/fcl/BackwardCompatibility.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/egordon/Workspace/dart/dart/collision/fcl/FCLTypes.hpp:37,
                 from /home/egordon/Workspace/dart/dart/collision/fcl/FCLCollisionDetector.hpp:38,
                 from /home/egordon/Workspace/dart/dart/collision/fcl/FCLCollisionDetector.cpp:33:
/home/egordon/Workspace/dart/dart/collision/fcl/BackwardCompatibility.hpp:49:10: fatal error: fcl/config.h: No such file or directory
 #include <fcl/config.h>
          ^~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [dart/CMakeFiles/dart.dir/build.make:1506: dart/CMakeFiles/dart.dir/collision/fcl/FCLCollisionDetector.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:342: dart/CMakeFiles/dart.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

`cmake` output:
```
-- ============================================
--                 DART 6.7.0
-- ============================================
-- Found CCD: /usr/include (found version "2.0") 
-- Could NOT find FCL (missing: FCL_INCLUDE_DIRS FCL_LIBRARIES) 
-- Found ASSIMP: /usr/include (found version "4.1.0") 
-- Performing Test ASSIMP_AISCENE_CTOR_DTOR_DEFINED
-- Performing Test ASSIMP_AISCENE_CTOR_DTOR_DEFINED - Success
-- Performing Test ASSIMP_AIMATERIAL_CTOR_DTOR_DEFINED
-- Performing Test ASSIMP_AIMATERIAL_CTOR_DTOR_DEFINED - Success
CMake Warning at cmake/DARTFindDependencies.cmake:103 (message):
  Octomap 1.9.0 is found, but Octomap 1.9.0 or greater is not supported yet.
  Please see 'https://github.com/dartsim/dart/issues/1078' for the details
Call Stack (most recent call first):
  CMakeLists.txt:137 (include)


-- Looking for ipopt - NOT found, to use dart-optimizer-ipopt, please install ipopt
-- Looking for nlopt - NOT found, to use dart-optimizer-nlopt, please install nlopt
-- Looking for pagmo - NOT found, to use dart-optimizer-pagmo, please install pagmo
-- Looking for ode - NOT found, to use dart-collision-ode, please install ode
-- Looking for Bullet - NOT found, to use dart-collision-bullet, please install libbullet-dev
-- Found FLANN: /usr/include (found version "1.9.1") 
-- Found TINYXML2: /usr/include (found version "7.0.1") 
-- Could NOT find OpenSceneGraph nor OpenThreads -- we will skip dart-gui-osg
If you believe you do have both OSG and OpenThreads installed, try setting OSG_DIR
-- Adding 8 components including the default component 'dart'
-- Adding 45 tests (comprehensive: 16, regression: 4, unit: 25)
-- Adding 16 examples
-- Adding 6 tutorials
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@
 * Build system
 
   * Changed to use GNUInstallDirs for install paths: [#1241](https://github.com/dartsim/dart/pull/1241)
+  * Fixed not failing for missing required dependencies: [#1250](https://github.com/dartsim/dart/pull/1250)
 
 #### Compilers Tested
 

--- a/cmake/Findassimp.cmake
+++ b/cmake/Findassimp.cmake
@@ -38,7 +38,7 @@ set(ASSIMP_VERSION ${PC_ASSIMP_VERSION})
 
 # Set (NAME)_FOUND if all the variables and the version are satisfied.
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(ASSIMP
+find_package_handle_standard_args(assimp
     FAIL_MESSAGE  DEFAULT_MSG
     REQUIRED_VARS ASSIMP_INCLUDE_DIRS ASSIMP_LIBRARIES
     VERSION_VAR   ASSIMP_VERSION)

--- a/cmake/Findccd.cmake
+++ b/cmake/Findccd.cmake
@@ -20,7 +20,7 @@
 find_package(ccd QUIET CONFIG)
 # Upstream provide ccd-config.cmake since 2.1.
 
-if(NOT CCD_FOUND)
+if(NOT CCD_FOUND AND NOT ccd_FOUND)
 
   find_package(PkgConfig QUIET)
 
@@ -63,7 +63,7 @@ if(NOT CCD_FOUND)
 
   # Set (NAME)_FOUND if all the variables and the version are satisfied.
   include(FindPackageHandleStandardArgs)
-  find_package_handle_standard_args(CCD
+  find_package_handle_standard_args(ccd
       FAIL_MESSAGE  DEFAULT_MSG
       REQUIRED_VARS CCD_INCLUDE_DIRS CCD_LIBRARIES
       VERSION_VAR   CCD_VERSION)

--- a/cmake/Findfcl.cmake
+++ b/cmake/Findfcl.cmake
@@ -57,7 +57,7 @@ set(FCL_VERSION ${PC_FCL_VERSION})
 
 # Set (NAME)_FOUND if all the variables and the version are satisfied.
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(FCL
+find_package_handle_standard_args(fcl
     FAIL_MESSAGE  DEFAULT_MSG
     REQUIRED_VARS FCL_INCLUDE_DIRS FCL_LIBRARIES
     VERSION_VAR   FCL_VERSION)

--- a/cmake/Findflann.cmake
+++ b/cmake/Findflann.cmake
@@ -37,7 +37,7 @@ set(FLANN_VERSION ${PC_FLANN_VERSION})
 
 # Set (NAME)_FOUND if all the variables and the version are satisfied.
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(FLANN
+find_package_handle_standard_args(flann
     FAIL_MESSAGE  DEFAULT_MSG
     REQUIRED_VARS FLANN_INCLUDE_DIRS FLANN_LIBRARIES
     VERSION_VAR   FLANN_VERSION)

--- a/cmake/Findtinyxml2.cmake
+++ b/cmake/Findtinyxml2.cmake
@@ -42,7 +42,7 @@ set(TINYXML2_VERSION ${PC_TINYXML2_VERSION})
 
 # Set (NAME)_FOUND if all the variables and the version are satisfied.
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(TINYXML2
+find_package_handle_standard_args(tinyxml2
     FAIL_MESSAGE  DEFAULT_MSG
     REQUIRED_VARS TINYXML2_INCLUDE_DIRS TINYXML2_LIBRARIES
     VERSION_VAR   TINYXML2_VERSION)

